ui/app/AppLayouts/Wallet/views/RecipientView.qml

```diff
@@ -0,0 +1,193 @@
+            }
+            else if(isENSName) {
+                d.uuid = Utils.uuid()
+                return root.resolveENS(address, uuid)
+            } else {
+                root.selectedRecipientAddress = ""
+                d.isBeingEvaluated = false
+            }
+        })
+
+        readonly property var accountsSelectedEntry: ModelEntry {
+            sourceModel: root.myAccountsModel
+            key: "address"
+            value: root.selectedRecipientAddress
+        }
+
+        readonly property var savedAddrSelectedEntry: ModelEntry {
+            sourceModel: root.savedAddressesModel
+            key: "address"
+            value: root.selectedRecipientAddress
+
+        }
+
+        function clearValues() {
+            root.selectedRecipientAddress = ""
+            root.selectedRecipientType = Helpers.RecipientAddressObjectType.Address
+        }
+    }
+
+    sourceComponent: root.selectedRecipientType === Helpers.RecipientAddressObjectType.SavedAddress ?
+                         savedAddressRecipient:
+                         root.selectedRecipientType === Helpers.RecipientAddressObjectType.Account ?
+                             myAccountRecipient:
+                             root.selectedRecipientType === Helpers.RecipientAddressObjectType.RecentsAddress ?
+                                 recentsRecipient : addressRecipient
+
+    Component {
+        id: savedAddressRecipient
+        SavedAddressListItem {
+            implicitWidth: parent.width
+            modelData: d.savedAddrSelectedEntry.item
+            radius: 8
+            clearVisible: true
+            color: Theme.palette.indirectColor1
+            sensor.enabled: false
+            subTitle:  {
+                if(!!modelData) {
+                    if (!!modelData && !!modelData.ens && modelData.ens.length > 0)
+                        return Utils.richColorText(modelData.ens, Theme.palette.directColor1)
+                    else
+                        return StatusQUtils.Utils.elideText(modelData.address,6,4)
+                }
+                return ""
+            }
+            onCleared: d.clearValues()
+        }
+    }
+
+    Component {
+        id: myAccountRecipient
+        SharedControls.WalletAccountListItem {
+            id: accountItem
+            readonly property var modelData: d.accountsSelectedEntry.item
+
+            name: !!modelData ? modelData.name : ""
+            address: !!modelData ? modelData.address : ""
+            emoji: !!modelData ? modelData.emoji : ""
+            walletColor: !!modelData ? Utils.getColorForId(modelData.colorId): ""
+            currencyBalance: !!modelData ? modelData.currencyBalance : ""
+            walletType: !!modelData ? modelData.walletType : ""
+            migratedToKeycard: !!modelData ? modelData.migratedToKeycard ?? false : false
+            accountBalance: !!modelData ? modelData.accountBalance : null
+
+            width: parent.width
```

Comment:
```suggestion
            width: ListView.view.width
```

Reply:
there is no ListView here, also this component will be reworked under  https://github.com/status-im/status-desktop/issues/16916